// @mui/x-charts/RadarChart/RadarSeriesPlot/RadarSeriesArea.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault").default;
var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard").default;
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.RadarSeriesArea = RadarSeriesArea;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _objectWithoutPropertiesLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutPropertiesLoose"));
var React = _interopRequireWildcard(require("react"));
var _propTypes = _interopRequireDefault(require("prop-types"));
var _useRadarSeriesData = require("./useRadarSeriesData");
var _getAreaPath = require("./getAreaPath");
var _jsxRuntime = require("react/jsx-runtime");
const _excluded = ["seriesId"];
function RadarSeriesArea(props) {
  const {
      seriesId
    } = props,
    other = (0, _objectWithoutPropertiesLoose2.default)(props, _excluded);
  const seriesCoordinates = (0, _useRadarSeriesData.useRadarSeriesData)(seriesId);
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(React.Fragment, {
    children: seriesCoordinates?.map(({
      seriesId: id,
      points,
      color
    }) => /*#__PURE__*/(0, _jsxRuntime.jsx)("path", (0, _extends2.default)({
      d: (0, _getAreaPath.getAreaPath)(points),
      fill: color,
      stroke: color,
      fillOpacity: 0.4
    }, other), id))
  });
}
process.env.NODE_ENV !== "production" ? RadarSeriesArea.propTypes = {
  // ----------------------------- Warning --------------------------------
  // | These PropTypes are generated from the TypeScript type definitions |
  // | To update them edit the TypeScript types and run "pnpm proptypes"  |
  // ----------------------------------------------------------------------
  /**
   * The id of the series to display.
   * If undefined all series are displayed.
   */
  seriesId: _propTypes.default.string
} : void 0;
