// @mui/x-charts/RadarChart/RadarSeriesPlot/RadarSeriesMarks.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault").default;
var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard").default;
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.RadarSeriesMarks = RadarSeriesMarks;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _objectWithoutPropertiesLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutPropertiesLoose"));
var React = _interopRequireWildcard(require("react"));
var _propTypes = _interopRequireDefault(require("prop-types"));
var _useRadarSeriesData = require("./useRadarSeriesData");
var _jsxRuntime = require("react/jsx-runtime");
const _excluded = ["seriesId"];
function RadarSeriesMarks(props) {
  const other = (0, _objectWithoutPropertiesLoose2.default)(props, _excluded);
  const seriesCoordinates = (0, _useRadarSeriesData.useRadarSeriesData)(props.seriesId);
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(React.Fragment, {
    children: seriesCoordinates?.map(({
      seriesId: id,
      points,
      color
    }) => /*#__PURE__*/(0, _jsxRuntime.jsx)("g", {
      children: points.map((point, index) => /*#__PURE__*/(0, _jsxRuntime.jsx)("circle", (0, _extends2.default)({
        cx: point.x,
        cy: point.y,
        r: 5,
        fill: color,
        stroke: color
      }, other), index))
    }, id))
  });
}
process.env.NODE_ENV !== "production" ? RadarSeriesMarks.propTypes = {
  // ----------------------------- Warning --------------------------------
  // | These PropTypes are generated from the TypeScript type definitions |
  // | To update them edit the TypeScript types and run "pnpm proptypes"  |
  // ----------------------------------------------------------------------
  /**
   * The id of the series to display.
   * If undefined all series are displayed.
   */
  seriesId: _propTypes.default.string
} : void 0;
