// @mui/x-charts/RadarChart/RadarGrid/RadarGrid.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault").default;
var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard").default;
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.RadarGrid = RadarGrid;
var React = _interopRequireWildcard(require("react"));
var _propTypes = _interopRequireDefault(require("prop-types"));
var _useRadarGridData = require("./useRadarGridData");
var _SharpRadarGrid = require("./SharpRadarGrid");
var _CircularRadarGrid = require("./CircularRadarGrid");
var _jsxRuntime = require("react/jsx-runtime");
function RadarGrid(props) {
  const {
    divisions = 5,
    shape = 'sharp'
  } = props;
  const gridData = (0, _useRadarGridData.useRadarGridData)();
  if (gridData === null) {
    return null;
  }
  const {
    center,
    corners,
    radius
  } = gridData;
  return shape === 'sharp' ? /*#__PURE__*/(0, _jsxRuntime.jsx)(_SharpRadarGrid.SharpRadarGrid, {
    divisions: divisions,
    corners: corners,
    center: center,
    radius: radius
  }) : /*#__PURE__*/(0, _jsxRuntime.jsx)(_CircularRadarGrid.CircularRadarGrid, {
    divisions: divisions,
    corners: corners,
    center: center,
    radius: radius
  });
}
process.env.NODE_ENV !== "production" ? RadarGrid.propTypes = {
  // ----------------------------- Warning --------------------------------
  // | These PropTypes are generated from the TypeScript type definitions |
  // | To update them edit the TypeScript types and run "pnpm proptypes"  |
  // ----------------------------------------------------------------------
  /**
   * The number of divisions in the radar grid.
   * @default 5
   */
  divisions: _propTypes.default.number,
  /**
   * The grid shape.
   * @default 'sharp'
   */
  shape: _propTypes.default.oneOf(['circular', 'sharp'])
} : void 0;
