// @mui/x-charts/RadarChart/RadarAxisHighlight/RadarAxisPointsHighlight.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard").default;
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.RadarAxisPointsHighlight = RadarAxisPointsHighlight;
var React = _interopRequireWildcard(require("react"));
var _styles = require("@mui/material/styles");
var _useRadarAxisHighlight = require("./useRadarAxisHighlight");
var _jsxRuntime = require("react/jsx-runtime");
/**
 * @ignore - internal component.
 */
function RadarAxisPointsHighlight(props) {
  const {
    classes
  } = props;
  const theme = (0, _styles.useTheme)();
  const data = (0, _useRadarAxisHighlight.useRadarAxisHighlight)();
  if (data === null) {
    return null;
  }
  const {
    center,
    series,
    points,
    radius,
    highlightedAngle,
    instance
  } = data;
  const [x, y] = instance.polar2svg(radius, highlightedAngle);
  return /*#__PURE__*/(0, _jsxRuntime.jsxs)("g", {
    className: classes?.root,
    children: [/*#__PURE__*/(0, _jsxRuntime.jsx)("path", {
      d: `M ${center.cx} ${center.cy} L ${x} ${y}`,
      stroke: (theme.vars || theme).palette.text.primary,
      strokeWidth: 3,
      className: classes?.line,
      pointerEvents: "none"
    }), points.map(({
      highlighted
    }, seriesIndex) => {
      return /*#__PURE__*/(0, _jsxRuntime.jsx)("circle", {
        fill: series[seriesIndex].color,
        cx: highlighted.x,
        cy: highlighted.y,
        r: 8,
        className: classes?.dot,
        pointerEvents: "none"
      }, series[seriesIndex].id);
    })]
  });
}
