// @mui/x-charts/esm/RadarChart/RadarGrid/SharpRadarGrid.js
import * as React from 'react';
import { useTheme } from '@mui/material/styles';
import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
/**
 * @ignore - internal component.
 */
export function SharpRadarGrid(props) {
  const {
    center,
    corners,
    divisions
  } = props;
  const theme = useTheme();
  const divisionRatio = Array.from({
    length: divisions
  }, (_, index) => (index + 1) / divisions);
  return /*#__PURE__*/_jsxs(React.Fragment, {
    children: [corners.map(({
      x,
      y
    }, i) => /*#__PURE__*/_jsx("path", {
      d: `M ${center.x} ${center.y} L ${x} ${y}`,
      stroke: (theme.vars || theme).palette.text.primary,
      strokeWidth: 1,
      fill: "none"
    }, i)), divisionRatio.map(ratio => /*#__PURE__*/_jsx("path", {
      d: `M ${corners.map(({
        x,
        y
      }) => `${center.x * (1 - ratio) + ratio * x} ${center.y * (1 - ratio) + ratio * y}`).join(' L ')} Z`,
      stroke: (theme.vars || theme).palette.text.primary,
      strokeWidth: 1,
      fill: "none"
    }, ratio))]
  });
}
