// @mui/x-charts/modern/RadarChart/RadarAxisHighlight/RadarAxisSliceHighlight.js
import * as React from 'react';
import { useRadarAxisHighlight } from "./useRadarAxisHighlight.js";
import { jsx as _jsx } from "react/jsx-runtime";
const params = {
  includesNeighbors: true
};
/**
 * @ignore - internal component.
 */
export function RadarAxisSliceHighlight(props) {
  const {
    classes
  } = props;
  const data = useRadarAxisHighlight(params);
  if (data === null) {
    return null;
  }
  const {
    center,
    series,
    points
  } = data;
  return /*#__PURE__*/_jsx("g", {
    className: classes?.root,
    children: series.map(({
      id,
      color
    }, seriesIndex) => {
      const {
        highlighted,
        next = highlighted,
        previous = highlighted
      } = points[seriesIndex];
      const ratioPrev = previous.r / (previous.r + highlighted.r);
      const ratioNext = next.r / (next.r + highlighted.r);
      return /*#__PURE__*/_jsx("path", {
        className: classes?.slice,
        fill: color,
        d: `M
 ${center.cx} ${center.cy}
 L ${previous.x * (1 - ratioPrev) + highlighted.x * ratioPrev} ${previous.y * (1 - ratioPrev) + highlighted.y * ratioPrev}
 L ${highlighted.x} ${highlighted.y}
 L ${next.x * (1 - ratioNext) + highlighted.x * ratioNext} ${next.y * (1 - ratioNext) + highlighted.y * ratioNext}
 Z
 `,
        pointerEvents: "none"
      }, id);
    })
  });
}
