// @mui/x-charts/modern/RadarChart/RadarAxisHighlight/RadarAxisPointsHighlight.js
import * as React from 'react';
import { useTheme } from '@mui/material/styles';
import { useRadarAxisHighlight } from "./useRadarAxisHighlight.js";
import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
/**
 * @ignore - internal component.
 */
export function RadarAxisPointsHighlight(props) {
  const {
    classes
  } = props;
  const theme = useTheme();
  const data = useRadarAxisHighlight();
  if (data === null) {
    return null;
  }
  const {
    center,
    series,
    points,
    radius,
    highlightedAngle,
    instance
  } = data;
  const [x, y] = instance.polar2svg(radius, highlightedAngle);
  return /*#__PURE__*/_jsxs("g", {
    className: classes?.root,
    children: [/*#__PURE__*/_jsx("path", {
      d: `M ${center.cx} ${center.cy} L ${x} ${y}`,
      stroke: (theme.vars || theme).palette.text.primary,
      strokeWidth: 3,
      className: classes?.line,
      pointerEvents: "none"
    }), points.map(({
      highlighted
    }, seriesIndex) => {
      return /*#__PURE__*/_jsx("circle", {
        fill: series[seriesIndex].color,
        cx: highlighted.x,
        cy: highlighted.y,
        r: 8,
        className: classes?.dot,
        pointerEvents: "none"
      }, series[seriesIndex].id);
    })]
  });
}
