// @mui/x-charts/modern/RadarChart/RadarGrid/RadarGrid.js
import * as React from 'react';
import PropTypes from 'prop-types';
import { useRadarGridData } from "./useRadarGridData.js";
import { SharpRadarGrid } from "./SharpRadarGrid.js";
import { CircularRadarGrid } from "./CircularRadarGrid.js";
import { jsx as _jsx } from "react/jsx-runtime";
function RadarGrid(props) {
  const {
    divisions = 5,
    shape = 'sharp'
  } = props;
  const gridData = useRadarGridData();
  if (gridData === null) {
    return null;
  }
  const {
    center,
    corners,
    radius
  } = gridData;
  return shape === 'sharp' ? /*#__PURE__*/_jsx(SharpRadarGrid, {
    divisions: divisions,
    corners: corners,
    center: center,
    radius: radius
  }) : /*#__PURE__*/_jsx(CircularRadarGrid, {
    divisions: divisions,
    corners: corners,
    center: center,
    radius: radius
  });
}
process.env.NODE_ENV !== "production" ? RadarGrid.propTypes = {
  // ----------------------------- Warning --------------------------------
  // | These PropTypes are generated from the TypeScript type definitions |
  // | To update them edit the TypeScript types and run "pnpm proptypes"  |
  // ----------------------------------------------------------------------
  /**
   * The number of divisions in the radar grid.
   * @default 5
   */
  divisions: PropTypes.number,
  /**
   * The grid shape.
   * @default 'sharp'
   */
  shape: PropTypes.oneOf(['circular', 'sharp'])
} : void 0;
export { RadarGrid };
